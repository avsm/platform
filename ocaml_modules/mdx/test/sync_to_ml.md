Mdx can also understand ocaml code blocks:


```ocaml file=sync_to_ml.ml,part=toto
# let x = 3;;
val x : int = 3
# let y = 4;;
val y : int = 4
# print_int x;;
3
- : unit = ()
# print_int y;;
4
- : unit = ()
```

```ocaml file=sync_to_ml.ml,part=zzz
```


```ocaml
# let x = 2;;
val x : int = 2
# print_int x;;
3
- : unit = ()
```
