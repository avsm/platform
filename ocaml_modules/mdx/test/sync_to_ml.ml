let do_hash file =
   In_channel.with_file file ~f:(fun ic ->
     let open Cryptokit in
     hash_channel (Hash.md5 ()) ic
     |> transform_string (Hexa.encode ())
     |> print_endline
   )
[@@@part "1"];;
let filename_param =
  let open Command.Param in
  anon ("filename" %: string)
[@@@part "toto"];;
let x = 34
let f = 42.3
let s = "toto"
let f x u = u x

let () =
  print_int x;
  print_float f
;;

[@@@part "zzz"];;
let () =
  print_string s
;;

[@@@part "42"];;

let () =
  f x print_int
