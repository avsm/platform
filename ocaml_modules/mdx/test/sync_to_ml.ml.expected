let () =
  ();
  ()
;;
[@@@part "toto"];;
let x = 3;;
let y = 4;;
print_int x;;
print_int y;;
[@@@part "zzz"];;
[@@@part "42"];;
let () =
  f x print_int
